app/components/channel_drawer/channels_list/filtered_list/index.js

```diff
@@ -25,15 +25,62 @@ const pastDirectMessages = createSelector(
     (directChannelsFromPreferences) => directChannelsFromPreferences.filter((d) => d.value === 'false').map((d) => d.name)
 );
 
+const getTeamProfiles = createSelector(
+    getProfilesInCurrentTeam,
+    (members) => {
+        return members.reduce((memberProfiles, member) => {
+            memberProfiles[member.id] = member;
+
+            return memberProfiles;
+        }, {});
+    }
+);
+
+function getGroupDetails(currentUserId, userIdsInChannels, profiles, groupChannels) {
+    return groupChannels.reduce((groupMemberDetails, channel) => {
+        if (!userIdsInChannels.hasOwnProperty(channel.id)) {
+            return groupMemberDetails;
+        }
+
+        const members = Array.from(userIdsInChannels[channel.id]).reduce((details, member) => {
+            if (member === currentUserId) {
+                return details;
+            }
+
+            const profile = profiles[member];
+            const username = `${details.username || ''} ${profile.username}`;
```

Comment:
I don't know if I like the idea of concatenating these all to make a search string. I feel like there might be a case like where someone searches for "bob ross" and you have two people in the group "bob" and "ross" and it returns the group as a result. That would require them to appear in that very specific order though, so maybe that's actually okay.

No functional changes for now, but maybe add a comment about how this works? It took me a few minutes to understand what this code was trying to accomplish and what "group details" meant